### ISSUE TITLE
Loconet LbServer TrackPower On Command is not Sent

### ISSUE THREAD

**Issue by DrEVILish** [traintastic/traintastic#143]
**Describe the bug**
Clicking Power On in the UI doesn't send LocoNet TrackPower On Command to the Server

**To Reproduce**
1. Create LocoNet Interface, set IP and Port
2. Connect
3. Click Power ON

**Expected behavior**
Send Track Power on Command via Loconet

**Screenshots**
<img width="994" alt="Screenshot 2024-06-19 at 19 27 29" src="https://github.com/traintastic/traintastic/assets/6287003/bcad565e-36de-4261-bd00-e5129caed09f">

**World export**
If applicable, please attach an export of the world that has the bug. (In traintastic-client: *File* -> *Export world*)

**Traintastic server:**
 - OS: macOS dev-0.3.0

**Traintastic client:**
 - OS: macOS dev-0.3.0

**Connected hardware**
RedHat++/DCC-EX

**Comment by reinder** [comment]
Can you enable LocoNet communication debug logging and post the log here, then we can see which loconet messages are sent and which aren't.

**Comment by DrEVILish** [comment]
`Log input changes` TICKED
`Log all communication` TICKED

Log messages:

```2024-06-19 21:40:32.352302 world N1013: Communication: disabled
2024-06-19 21:40:33.218668 world N1015: Power: off
2024-06-19 21:40:34.819788 world N1017: Stopped
2024-06-19 21:41:17.319218 world N1012: Communication: enabled
2024-06-19 21:41:17.397831 loconet_1 D2001: TX: OPC_RQ_SL_DATA slot=1 [BB 01 00 45]
2024-06-19 21:41:29.900295 world N1014: Power: on```
<img width="909" alt="Screenshot 2024-06-19 at 21 50 00" src="https://github.com/traintastic/traintastic/assets/6287003/94ed4e84-8746-4b59-b371-0c10167f9278">

Also doesn't transmit Throttle change data

LocoNetViewer from RedHat++ when using a 3rd Party WiThrottle Interface: 
```
...OPC_LOCO_DIRF 0xA1, 0x01, 0x30, 0x6F -- Set function group 1 for slot 1 to 0x30
OPC_LOCO_SPD 0xA0, 0x01, 0x1E, 0x40 -- Set Loco speed for slot 1 to 30
OPC_LOCO_DIRF 0xA1, 0x01, 0x30, 0x6F -- Set function group 1 for slot 1 to 0x30
OPC_LOCO_SPD 0xA0, 0x01, 0x25, 0x7B -- Set Loco speed for slot 1 to 37
OPC_LOCO_DIRF 0xA1, 0x01, 0x30, 0x6F -- Set function group 1 for slot 1 to 0x30
OPC_LOCO_SPD 0xA0, 0x01, 0x00, 0x5E -- Set Loco speed for slot 1 to 0...
```
Traintastic Logs nothing to the console during this time.

**Comment by reinder** [comment]
When Traintastic connects to a LocoNet command station it starts by fetching all slots, so it knows the actual status. All these messages including power/throttle message are queued and sent one, wait for reply, sent next etc.

Based on the log it seems that there is no reply on the slot read request (do you see a timeout error?), therefor it doesn't sent the next one.

When connecting to a LocoNet command station you normally see something like:
```
2024-06-19 23:22:04.218279 world N1012: Communication: enabled
2024-06-19 23:22:04.221625 loconet_1 D2001: TX: OPC_RQ_SL_DATA slot=1 [BB 01 00 45]
2024-06-19 23:22:04.221870 loconet_1 D2002: RX: OPC_RQ_SL_DATA slot=1 [BB 01 00 45]
2024-06-19 23:22:04.221981 loconet_1 D2002: RX: OPC_SL_RD_DATA [E7 0E 01 00 00 00 00 00 00 00 00 00 00 17]
2024-06-19 23:22:04.222146 loconet_1 D2001: TX: OPC_RQ_SL_DATA slot=2 [BB 02 00 46]
2024-06-19 23:22:04.222271 loconet_1 D2002: RX: OPC_RQ_SL_DATA slot=2 [BB 02 00 46]
2024-06-19 23:22:04.222399 loconet_1 D2002: RX: OPC_SL_RD_DATA [E7 0E 02 00 00 00 00 00 00 00 00 00 00 14]
2024-06-19 23:22:04.222569 loconet_1 D2001: TX: OPC_RQ_SL_DATA slot=3 [BB 03 00 47]
2024-06-19 23:22:04.222699 loconet_1 D2002: RX: OPC_RQ_SL_DATA slot=3 [BB 03 00 47]
2024-06-19 23:22:04.222821 loconet_1 D2002: RX: OPC_SL_RD_DATA [E7 0E 03 00 00 00 00 00 00 00 00 00 00 15]
--- CUT: same for slot 4..117 ---
2024-06-19 23:22:04.274304 loconet_1 D2002: RX: OPC_SL_RD_DATA [E7 0E 75 00 00 00 00 00 00 00 00 00 00 63]
2024-06-19 23:22:04.274621 loconet_1 D2001: TX: OPC_RQ_SL_DATA slot=118 [BB 76 00 32]
2024-06-19 23:22:04.274880 loconet_1 D2002: RX: OPC_RQ_SL_DATA slot=118 [BB 76 00 32]
2024-06-19 23:22:04.275133 loconet_1 D2002: RX: OPC_SL_RD_DATA [E7 0E 76 00 00 00 00 00 00 00 00 00 00 60]
2024-06-19 23:22:04.275386 loconet_1 D2001: TX: OPC_RQ_SL_DATA slot=119 [BB 77 00 33]
2024-06-19 23:22:04.275639 loconet_1 D2002: RX: OPC_RQ_SL_DATA slot=119 [BB 77 00 33]
2024-06-19 23:22:04.275891 loconet_1 D2002: RX: OPC_SL_RD_DATA [E7 0E 77 00 00 00 00 00 00 00 00 00 00 61]
```

**Comment by gfgit** [comment]
Is this a protocol limitation of not sending multiple messages before receiving a reply or a behaviour specific to Traintastic?
It may be worth adding a timeout after which it moves on to the next queued message.

**Comment by DrEVILish** [comment]
I would agree but I'm not seeing any Loconet signals being received at the "Loconet server"

Current timeout set to 1000ms

**Comment by reinder** [comment]
> Is this a protocol limitation of not sending multiple messages before receiving a reply or a behaviour specific to Traintastic? It may be worth adding a timeout after which it moves on to the next queued message.

LocoNet protocol limitation, steps are:
- Sent message
- Wait for message echo
- Wait for reply (if message has a reply, can be determined by a bit of the OpCode)

There is also an echo timeout..but I don't see that logged. On timeout Traintastic will put the interface in error state, sending the next message doesn't make much sense as the bus is already not responding so there is clearly something wrong.

I'd expect there is a difference in the LBserver protocol implementation between Traintastic and RedHat++, maybe it is line ending realted. @DrEVILish can you try to use `"\r\n"` instead of `"\n"` at [lbserveriohandler.cpp line 38](https://github.com/traintastic/traintastic/blob/master/server/src/hardware/protocol/loconet/iohandler/lbserveriohandler.cpp#L38)

**Comment by DrEVILish** [comment]
[Hans Tanner IoTT Stick LbServer](https://github.com/tanner87661/IoTTStick/blob/master/Sketchbook/libraries/IoTT_lbServer/src/IoTT_lbServer.cpp) Uses "\r\n" as the line end.
I will build and test, I'll get back with the results shortly.

**Comment by DrEVILish** [comment]
Still not getting a reply, "IoTT LbServer" Thinks that it has sent the reply but traintastic isn't registering the reply. - macOS & Debian - server 

<img width="1595" alt="Screenshot 2024-06-21 at 12 02 27" src="https://github.com/traintastic/traintastic/assets/6287003/753697fa-776b-4ec8-82f8-017e7d989e6b">

**Comment by reinder** [comment]
@DrEVILish with your fix #147 does that resolve the communication issue?

**Comment by DrEVILish** [comment]
Yes, there is a new bug, #148, but I believe it's unrelated. I'll try to motor through it when I get some time.
